lib/whitehall_importer/create_file_attachment_revision.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+module WhitehallImporter
+  class CreateFileAttachmentRevision
+    def self.call(*args)
+      new(*args).call
+    end
+
+    def initialize(whitehall_file_attachment, existing_filenames = [])
+      @whitehall_file_attachment = whitehall_file_attachment
+      @existing_filenames = existing_filenames
+    end
+
+    def call
+      check_whitehall_file_attachment_type
+      check_file_requirements
+      blob_revision = create_blob_revision
+
+      FileAttachment::Revision.create!(
+        blob_revision: blob_revision,
+        file_attachment: FileAttachment.create!,
+        metadata_revision: FileAttachment::MetadataRevision.create!(
+          title: whitehall_file_attachment["title"],
+        ),
+      )
+    end
+
+  private
+
+    attr_reader :whitehall_file_attachment, :existing_filenames
+
+    def decorated_file
+      @decorated_file ||= AttachmentFileDecorator.new(download_file, unique_filename)
+    end
+
+    def unique_filename
+      @unique_filename ||= UniqueFilenameService.call(
+        existing_filenames,
+        File.basename(whitehall_file_attachment["url"]),
+      )
+    end
+
+    def download_file
+      file = URI.parse(whitehall_file_attachment["url"]).open
+      if file.is_a?(StringIO)
+        # files less than 10 KB return StringIO (we have to manually cast to a tempfile)
+        Tempfile.new.tap { |tmp| File.write(tmp.path, file.string) }
+      else
+        file
+      end
```

Comment:
Should we move this dealing with tempfiles somewhere else since we've got the same logic in this and for images. Alternatively is there a nicer way where can avoid the need for this StringIO handling? it might even be worth checking if this stuff is needed for attachment as we might be able to just work with a stringio instance within the decorated class.

Reply:
Did this - `Should we move this dealing with tempfiles somewhere else since we've got the same logic in this and for images`

Turns out we don't even need the StringIO bit as you though, tested in integration importing a 2kb -> StringIO file and imported successfully! 👍